### PR TITLE
add unittests for LAMMPS pair deepmd

### PR DESCRIPTION
**Pull request by njzjz**

### CODE REVIEW

**Comment by njzjz** [comment]
This PR is ok. More tests can be added in the coming PRs.

image

File: source/lmp/tests/test_lammps.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    lammps.timestep(0.0005)
+    lammps.fix("1 all nve")
+    yield lammps
+
+
+def test_pair_deepmd(lammps):
+    lammps.pair_style("deepmd {}".format(pb_file.resolve()))
+    lammps.pair_coeff("* *")
+    lammps.run(1)
+    assert lammps.eval("pe") != 0.
```
Comment (wanghan-iapcm) [inline source/lmp/tests/test_lammps.py:40]: write here the expected potential energy value?

File: source/lmp/tests/test_lammps.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+def test_pair_deepmd_model_devi(lammps):
```
Comment (wanghan-iapcm) [inline source/lmp/tests/test_lammps.py:43]: check the expected model deviation?

File: source/lmp/tests/test_lammps.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+def test_pair_deepmd_model_devi_atomic_relative_v(lammps):
+    relative = 1.0
+    lammps.pair_style("deepmd {} {} out_file {} out_freq 1 atomic relative_v {}".format(pb_file.resolve(), pb_file2.resolve(), md_file.resolve(), relative))
+    lammps.pair_coeff("* *")
+    lammps.run(0)
+    assert lammps.eval("pe") == pytest.approx(expected_e)
+    for ii in range(6):
+        assert lammps.atoms[ii].force == pytest.approx(expected_f[ii])
+    #md = np.loadtxt(md_file.resolve())
+    # TODO: how to get the virial?
```
Comment (wanghan-iapcm) [inline source/lmp/tests/test_lammps.py:120]: What I did is `compute pressure` and then subtract the kinetic contribution.

Comment (wanghan-iapcm) [reply source/lmp/tests/test_lammps.py:120]: Another option may be [using `compute stress/atom NULL virial`](https://docs.lammps.org/compute_stress_atom.html), but I have never tried.

Comment (njzjz) [reply source/lmp/tests/test_lammps.py:120]: I got all zero values from `virial all stress/atom NULL virial`. There may be something else wrong.
```py
    lammps.compute("virial all stress/atom NULL virial")
    lammps.dump("1 all custom 1 dump id c_virial[1] c_virial[2] c_virial[3]")
```

```dump
ITEM: TIMESTEP
0
ITEM: NUMBER OF ATOMS
6
ITEM: BOX BOUNDS xy xz yz pp pp pp
0.0000000000000000e+00 1.3000000000000000e+01 0.0000000000000000e+00
0.0000000000000000e+00 1.3000000000000000e+01 0.0000000000000000e+00
0.0000000000000000e+00 1.3000000000000000e+01 0.0000000000000000e+00
ITEM: ATOMS id c_virial[1] c_virial[2] c_virial[3]
1 -0 -0 -0
2 -0 -0 -0
3 -0 -0 -0
4 -0 -0 -0
5 -0 -0 -0
6 -0 -0 -0
```

Comment (njzjz) [reply source/lmp/tests/test_lammps.py:120]: I see. `compute centroid/stress/atom` should be used instead...

Comment (njzjz) [reply source/lmp/tests/test_lammps.py:120]: Really weird, I got
```
ITEM: TIMESTEP
0
ITEM: NUMBER OF ATOMS
6
ITEM: BOX BOUNDS xy xz yz pp pp pp
0.0000000000000000e+00 1.3000000000000000e+01 0.0000000000000000e+00
0.0000000000000000e+00 1.3000000000000000e+01 0.0000000000000000e+00
0.0000000000000000e+00 1.3000000000000000e+01 0.0000000000000000e+00
ITEM: ATOMS id c_virial[1] c_virial[2] c_virial[3] c_virial[4] c_virial[5] c_virial[6] c_virial[7] c_virial[8] c_virial[9]
1 -466591 -505066 -414063 -80970.2 452837 166487 -60892.5 444864 169974
2 -500155 -61207.7 -339939 66365 427598 -103312 135917 404495 -43696.3
3 -138177 -319263 -235570 -205665 224036 289084 -289945 245113 225895
4 -415602 -160778 -376371 -66875.1 54457.6 243892 -67876.8 54255.9 244443
5 1524.56 -153220 -381473 965.674 -12397.8 240780 -26327.3 31935.1 242059
6 -370016 -54260 -97381.6 -129762 161594 73006.7 -106816 127462 71263
```

Comment (njzjz) [reply source/lmp/tests/test_lammps.py:120]: The unit is different..

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**